prow/git/v2/interactor.go

```diff
@@ -278,6 +282,32 @@ func (i *interactor) FetchRef(refspec string) error {
 	return nil
 }
 
+// FetchFromRemote fetches all update from a specific remote and branch and leaves it as FETCH_HEAD.
+func (i *interactor) FetchFromRemote(remote RemoteResolver, branch string) error {
+	r, err := remote()
+	if err != nil {
+		return fmt.Errorf("couldn't get remote: %v", err)
+	}
+
+	// Parse the remote's URI to censor any existing tokens.
+	u, err := url.Parse(r)
+	if err != nil {
+		return fmt.Errorf("couldn't parse remote url: %v", err)
+	}
+
+	if token, ok := u.User.Password(); ok {
+		i.logger.Logger.SetFormatter(logrusutil.NewCensoringFormatter(logrus.StandardLogger().Formatter, func() sets.String {
```

Comment:
As written this code _overwrites_ the other formatters.

Reply:
/lgtm cancel
Missed this

Reply:
We can remove this if we rely on user to censor the secrets.